pkg/operatorsource/configuring.go

```diff
@@ -15,6 +16,13 @@ import (
 // NewConfiguringReconciler returns a Reconciler that reconciles
 // an OperatorSource object in "Configuring" phase.
 func NewConfiguringReconciler(logger *log.Entry, datastore datastore.Writer, client client.Client) Reconciler {
+	return NewReconcilerWithMarketplaceClient(logger, datastore, interface_client.NewClient(client))
+}
+
+// NewReconcilerWithMarketplaceClient returns a configuring Reconciler
+// that reconciles an OperatorSource object in "Configuring" phase.
+// It takes an additional CatalogSourceConfig object as input.
+func NewReconcilerWithMarketplaceClient(logger *log.Entry, datastore datastore.Writer, client interface_client.Client) Reconciler {
```

Comment:
...and refer to it in the function name as `MarketplaceClient`. This is inconsistent. Moreover this is not a `MarketplaceClient` but a generic kube client. You should also call out that this only being used for unit testing. Also explain why you need this wrapper.